test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,323 @@
+
+    kernel_signature = inspect.signature(kernel)
+    kernel_params = list(kernel_signature.parameters.values())[1:]
+
+    # We filter out metadata that is implicitly passed to the dispatcher through the input datapoint, but has to be
+    # explicit passed to the kernel.
+    kernel_params = [param for param in kernel_params if param.name not in datapoint_type.__annotations__.keys()]
+
+    dispatcher_params = iter(dispatcher_params)
+    for dispatcher_param, kernel_param in zip(dispatcher_params, kernel_params):
+        try:
+            # In general, the dispatcher parameters are a superset of the kernel parameters. Thus, we filter out
+            # dispatcher parameters that have no kernel equivalent while keeping the order intact.
+            while dispatcher_param.name != kernel_param.name:
+                dispatcher_param = next(dispatcher_params)
+        except StopIteration:
+            raise AssertionError(
+                f"Parameter `{kernel_param.name}` of kernel `{kernel.__name__}` "
+                f"has no corresponding parameter on the dispatcher `{dispatcher.__name__}`."
+            ) from None
+
+        assert dispatcher_param == kernel_param
+
+
+def _check_dispatcher_datapoint_signature_match(dispatcher):
+    dispatcher_signature = inspect.signature(dispatcher)
+    dispatcher_params = list(dispatcher_signature.parameters.values())[1:]
+
+    datapoint_method = getattr(datapoints._datapoint.Datapoint, dispatcher.__name__)
+    datapoint_signature = inspect.signature(datapoint_method)
+    datapoint_params = list(datapoint_signature.parameters.values())[1:]
+
+    # Because we use `from __future__ import annotations` inside the module where `datapoints._datapoint` is
+    # defined, the annotations are stored as strings. This makes them concrete again, so they can be compared to the
+    # natively concrete dispatcher annotations.
+    datapoint_annotations = get_type_hints(datapoint_method)
+    for param in datapoint_params:
+        param._annotation = datapoint_annotations[param.name]
+
+    assert dispatcher_params == datapoint_params
+
+
+def check_dispatcher_signatures_match(dispatcher, *, kernel, datapoint_type):
+    _check_dispatcher_kernel_signature_match(dispatcher, kernel=kernel, datapoint_type=datapoint_type)
+    _check_dispatcher_datapoint_signature_match(dispatcher)
+
+
+def check_transform():
+    pass
+
+
+class TestResize:
+    @pytest.mark.parametrize("size", [(11, 17), (15, 13)])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("batch_dims", DEFAULT_BATCH_DIMS)
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_image_tensor(self, size, antialias, batch_dims, dtype, device):
+        image = make_image(size=(14, 16), extra_dims=batch_dims, dtype=dtype, device=device)
+        check_kernel(F.resize_image_tensor, image, size=size, antialias=antialias)
```

Comment:
The more parametrization we add, the more I think having a single `check_kernel()` entrypoint is overkill.

Do we really need to check `jit vs eager` on all of those batch dims? With all those dtypes?
Do we really need to check `batch vs unbatched` on both devices, all these dtypes, with and without antialias?
Do we really need to check `cpu vs cuda` on all those batch dims?

I know we want to make sure we have good coverage, but are we going overboard with all those checks?